libraries/Microsoft.Bot.Builder.Core/MiddlewareSet.cs

```diff
@@ -39,16 +38,12 @@ public MiddlewareSet Use(IMiddleware middleware)
         /// </summary>
         public async Task ReceiveActivityWithStatus(IBotContext context, Func<IBotContext, Task> callback)
         {
-            await ReceiveActivityInternal(context, _middleware, callback).ConfigureAwait(false);
+            await ReceiveActivityInternal(context, callback).ConfigureAwait(false);
         }
 
-        private static async Task ReceiveActivityInternal(
-            IBotContext context, IEnumerable<IMiddleware> middleware, Func<IBotContext, Task> callback)
+        private Task ReceiveActivityInternal(IBotContext context, Func<IBotContext, Task> callback, int nextMiddlewareIndex = 0)
```

Comment:
I'm surprised you removed the Async marker from this method and moved it to just a Task. 

What's your thinking here? Could you articulate this in a detailed comment?

Reply:
Sure, when you ask for a comment, do you mean here on the PR or adding something to the code? In case you mean here in the PR, here goes:

There's nothing too "deep" to the removal of `async`/`await` except that it's simply not needed here IMHO. This is not a complicated flow and because performance is of utmost importance here it's better not to introduce the `async` state machine the compiler will generate into the mix.

Truth be told this would just be step one of improving the performance here for me. This code path, the path that every single activity execution passes through, should be as tight as possible and there are at least a few more things we can do to improve it. When you think about it, it's kind of wasteful to constantly pay the price for what is effectively a static path of execution once the pipeline is "baked". Ultimately we should be to build this execution path out one time and simply reuse it over and over again to remove (almost) all allocations in the hot path.